### PR TITLE
Dismiss the independent time conductor popup on unmount

### LINKED ISSUE #6842: Independent time conductor popup is not dismissed on navigating to a different object
#### Summary
Enabling the ITC and changing settings, then navving to another view (such as an Overlay Plot) without closing the itc popup does not dismiss it.

#### Expected vs Current Behavior
The ITC popup should get dismissed if you navigate away from the currently viewed object

#### Steps to Reproduce
1. Enable the ITC in a view
2. Click on the ITC settings to bring up the popup
3. Navigate to a different item by choosing it in the tree
4. Observe that the ITC popup remains open

#### Environment

* Open MCT Version: 3.0.0
* Deployment Type: viper/testathon
* OS: All
* Browser: All

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

### PR DESCRIPTION
**Pull request by shefalijoshi**
Closes #6842

### Describe your changes:
Don't set conductor popup to null unless the view is being destroyed

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [ ] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?
* [x] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js [hunk 1]
```diff
@@ -31,14 +31,17 @@ export default {
...
+      document.body.removeChild(this.conductorPopup);
+      this.conductorPopup = null;
+    }
+  },
   mounted() {
     this.positionBox = debounce(raf(this.positionBox), 250);
     this.timeConductorOptionsHolder = this.$el;
     this.timeConductorOptionsHolder.addEventListener('click', this.showPopup);
   },
-  beforeDestroy() {
```
Comment (ozyx) [inline src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js:39]: woah! nice catch-- I didn't realize we still had `beforeDestroy()` hooks in the codebase but it looks like there are a few left behind. Let's update the others to `beforeUnmounted()` while we're here

**Comment by ozyx** [comment]
Navigating away from a view with an ITC on it is giving me this error:

image

**Review by ozyx — CHANGES_REQUESTED** [review]
Navigating away from a view with an ITC on it is giving me this error:

image

File: src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js [hunk 2]
```diff
@@ -31,14 +31,17 @@ export default {
       conductorPopup: null
     };
   },
+  beforeUnmount() {
+    if (this.conductorPopup.parentNode === document.body) {
```
Comment (ozyx) [inline src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js:35]: I think we just need a null check here

Comment (shefalijoshi) [reply src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js:35]: I don't think so. We would never be in a state where `this.conductorPopUp` would be null when we get to this code.

Comment (ozyx) [reply src/plugins/timeConductor/independent/independentTimeConductorPopUpManager.js:35]: If you navigate away from a displayLayout with an Overlay Plot in it without opening the Overlay Plot's ITC you will get into that state and it crashes navigation, see my review comment

**Review by ozyx — APPROVED** [review]
LGTM

**Pull request merged**